File: frontend/src/types/notification.ts
export type NotificationType =
  | 'BUDGET_WARNING'   // 예산 경고
  | 'BENEFIT_DEADLINE' // 혜택 마감
  | 'COMMENT'          // 댓글 알림

export interface Notification {
  id: number
  type: NotificationType
  title: string
  message: string
  linkUrl: string | null
  isRead: boolean
  createdAt: string
}

export interface UnreadCountResponse {
  count: number
}

File: backend/src/main/java/com/cashbook/domain/notification/entity/Notification.java
package com.cashbook.domain.notification.entity;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;

import java.time.OffsetDateTime;

/**
 * 알림 엔티티.
 * updated_at, deleted_at 없음 (is_read 업데이트만 발생).
 * type: BUDGET_ALERT | BENEFIT_DEADLINE | COMMUNITY_COMMENT
 */
@Getter
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Notification {
    private Long id;
    private Long userId;
    private String type;
    private String title;
    private String message;
    private boolean isRead;
    private String link;
    private OffsetDateTime createdAt;
}
